test/lib/middleware.test.ts

```diff
@@ -265,10 +293,26 @@ describe("Middleware Functions", () => {
     });
 
     it("should call the cache with the correct id", () => {
-      cacheMock.get.mockReturnValue({});
+      const registration = new Registration();
+      cacheMock.get.mockReturnValue(registration);
 
-      expect(getCache(id, cacheMock)).toStrictEqual({});
+      expect(getCache(id, cacheMock)).toStrictEqual(registration);
       expect(cacheMock.get).toHaveBeenCalledWith(id);
     });
```

Comment:
Discussed that mocking Registration is quite a big task and typescript won't really allow it to be done partially, so instead, I'll revert this change and use [type assertions](https://basarat.gitbook.io/typescript/type-system/type-assertion) to make typescript (?) happy.
![image](https://user-images.githubusercontent.com/32230328/115364118-f7e68200-a1ba-11eb-8696-39747a30ce32.png)

Reply:
Nice one :D Yep, it's Typescript that's worrying about the type of `{}`.  As the behaviour under test is `getCache()`, and that behaviour doesn't collaborate with the `Registration` argument, it's OK to bypass TypeScript's typing system to avoid having to mock the entire `Registration` object.  👍